### DISCUSSION TITLE
Pokemon Netplay for trading/battles doesn't work

### DISCUSSION THREAD

**Discussion by Nereushq** [OnionUI/Onion#1316] [Support]
Hello,

I treated myself to a Miyoo Mini Plus v3 about a month ago. My girlfriend saw it and wanted one too two weeks ago.

Both Miyoo's have:
Mode MY354
Version 202306282128
Onion OS 4.2.3

Supernintendo games work via Netplay. There are 1-2x connection interruptions but after that it works.

Now we wanted to trade Pokemon like we used to. We have Pokemon Gold and Silver for the GBC. But unfortunately that doesn't work at all.
There are a lot of connection interruptions, but once the connection is established, we are redirected to the main Pokemon screen and are supposed to create new savegames. Unfortunately our save is not used and it often freezes.

What have we already done?
-TGB Dual set as game core.
-Retroachievments are disabled.
-The TCP ports 55435 are released in the router.
-The saves are named the same as the game files.

We also tried this following the instructions: https://github.com/OnionUI/Onion/pull/1007

Stand in front of the Pokemon Center, save, exit and start Netplay.

Many instructions state that you should activate Pokemon in the Package Manager under Apps, Easy Netplay and Easy Netplay. Unfortunately we don't have that.

How do we start?
We go to the Rom press y. Then we go to Netplay (core supported: tgbdual), Host/Join a session...., Easy Netplay - Pokemon Trade/Battle

Do you want pictures, videos etc? I'll give you everything so that it might work.

Thanks for the help

**Comment by schmurtzm** [comment]
Hi, Many guides are deprecated since the final release of Onion 4.2.0 : if you use GLO menu it simplify Netplay experience compared to previously,for example all the manual configurations described by Russ (RetroGameCorp) are not necessary anymore : 

- you don't have to install "Easy Netplay" anymore from Package Manager (it is automatically proposed in GB section for MM+ only)
- you don't have to modify your current core
- you don't have to copy the save from Gambatte
- you don't have to configure TGBDual 

Concerning the connection stability, if you use "Easy Netplay" the distance between the two Miyoo Mini + must be short to avoid disconnections.

What I would do in your case :  
- Enable "Enable Logging" in Tweak app and reboot
- Put back Gambatte as default emulator and try if your save is OK in one player mode.
- Remove Pokemon TGB Dual saves located in "Saves/CurrentProfile/saves/TGB Dual" (they will be copied automatically by the script)
- restore TGB Dual options with Onion defaults
- Use GLO menu and try again (keep the distance short)
- Send a zipped file with the content of `.tmp_update/logs` for both consoles for analysis

**Comment by Nereushq** [comment]
Many thanks for your help.

Oh, those were all old instructions.

Unfortunately, my girlfriend's Miyoo Mini + memory card broke in the meantime.

We have now done everything according to your “instructions” and it works.

We always need 2-3 attempts until the connection is actually established... but it works really well after that.

Many thanks for your response

edit:

Can you also trade/battle Pokemon with Onion OS and Windows Retroarch?
